java/BUILD.bazel

```diff
@@ -91,6 +91,9 @@ define_java_module(
         "@maven//:commons_io_commons_io",
         "@maven//:de_ruedigermoeller_fst",
         "@maven//:net_java_dev_jna_jna",
+        "@maven//:org_apache_arrow_arrow_memory_core",
```

Comment:
It seems to be the place of [`runtime`](https://github.com/ray-project/ray/pull/35110/files/7c486ca3ee99ceb1448b08b46518e0dfcf54929f#diff-75e5d868c1627f70d4ec06eb8b017335b3991865567e366c71b331b057dbfcbeR61) already, or maybe I didn't get your pont.

Reply:
@Deegue My fault. I saw a wrong module.

Reply:
Added back `nogil` to all `write_to` functions.

However, for function [L567](https://github.com/ray-project/ray/pull/35110/files#diff-6eb0b173395e77eb7364c2f68f0e4ab2abdd1875e73aeb420159d1149721af13R567) we have to wrap the code block with `with gil` since pyarrow functions are used.